hs/src/Reach/AST.hs

```diff
@@ -240,6 +240,21 @@ data SLVal
   | SLV_Kwd SLKwd
   deriving (Eq, Generic, NFData, Show)
 
+slval_srcloc :: SLVal -> Maybe SrcLoc
+slval_srcloc = \case
+  SLV_Null a _ -> Just a
+  SLV_Bool a _ -> Just a
+  SLV_Int a _ -> Just a
+  SLV_Bytes a _ -> Just a
+  SLV_Array a _ _ -> Just a
+  SLV_Tuple a _ -> Just a
+  SLV_Object a _ _ -> Just a
+  SLV_Clo a _ _ _ _ -> Just a
+  SLV_Data a _ _ _ -> Just a
+  SLV_DLVar (DLVar a _ _ _) -> Just a
+  SLV_Participant a _ _ _ -> Just a
+  _ -> Nothing
+
```

Comment:
I actually recently added a type class for this: `SrcLocOf`

Reply:
I saw that, but was concerned about the type signature since a lot of `SLVal`s don't have `SrcLoc`. What are your thoughts about extending the signature to include a default param? I could leave it to the `SLVal` implementation to return an empty position, but I could provide a 'closer' location as a default. 

Also, I'll try adding some documentation for `match`

Reply:
I think it would be to return `srcloc_builtin` and check for it in this call site... even though that's generally bad, I think it's okay in this case.